rpcserver.go

```
@@ -1079,12 +1079,140 @@ func (r *rpcServer) VerifyProof(ctx context.Context,
+			Index: decodedProof.InclusionProof.OutputIndex,
+		}
+		txMerkleProof  = decodedProof.TxMerkleProof
+		inclusionProof = decodedProof.InclusionProof
+		splitRootProof = decodedProof.SplitRootProof
+	)
+
+	var txMerkleProofBuf bytes.Buffer
+	if err := txMerkleProof.Encode(&txMerkleProofBuf); err != nil {
+		return nil, fmt.Errorf("unable to encode serialized Bitcoin "+
+			"merkle proof: %w", err)
+	}
+
+	var inclusionProofBuf bytes.Buffer
+	if err := inclusionProof.Encode(&inclusionProofBuf); err != nil {
+		return nil, fmt.Errorf("unable to encode inclusion proof: %w",
+			err)
+	}
+
+	var exclusionProofs [][]byte
+	for _, exclusionProof := range decodedProof.ExclusionProofs {
+		var exclusionProofBuf bytes.Buffer
+		if err := exclusionProof.Encode(&exclusionProofBuf); err != nil {
+			return nil, fmt.Errorf("unable to encode exclusion "+
+				"proofs: %w", err)
+		}
+		exclusionProofBytes := exclusionProofBuf.Bytes()
+
+		exclusionProofs = append(exclusionProofs, exclusionProofBytes)
+	}
+
+	var splitRootProofBuf bytes.Buffer
+	if splitRootProof != nil {
+		if err := splitRootProof.Encode(&splitRootProofBuf); err != nil {
+			return nil, fmt.Errorf("unable to encode split root proof: %w",
+				err)
+		}
+	}
+
+	rpcAsset, err = r.marshalChainAsset(ctx, &tapdb.ChainAsset{
+		Asset:             &finalAsset,
+		AnchorTx:          &decodedProof.AnchorTx,
+		AnchorTxid:        decodedProof.AnchorTx.TxHash(),
+		AnchorBlockHash:   decodedProof.BlockHeader.BlockHash(),
+		AnchorOutpoint:    anchorOutpoint,
+		AnchorInternalKey: decodedProof.InclusionProof.InternalKey,
+	}, withPrevWitnesses)
+	if err != nil {
+		return nil, err
+	}
+
+	if withMetaReveal {
+		if len(rpcAsset.AssetGenesis.MetaHash) > 0 {
+			assetMeta, err := r.FetchAssetMeta(ctx, &taprpc.FetchAssetMetaRequest{
+				Asset: &taprpc.FetchAssetMetaRequest_MetaHash{
+					MetaHash: rpcAsset.AssetGenesis.MetaHash,
+				},
+			})
+			if assetMeta != nil && err == nil {
```

I don't think we should swallow the error here. If we specify with meta reveal but cannot fetch the meta information (because we might not have it), we should error out instead.

---

makes sense, fewer assumptions for the user